src/classes/STG_PanelPrimaryContactRoleMerge_CTRL.cls

```diff
@@ -0,0 +1,64 @@
+    
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+    
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+ 
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT 
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS 
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2018
+* @group Settings
+* @description Controller that runs the Batch Class that removes the duplicate primary OCRs.
+*/
+public with sharing class STG_PanelPrimaryContactRoleMerge_CTRL extends STG_Panel {
+
+    /*********************************************************************************************************
+    * @description The panel's constructor.
+    */
+    public STG_PanelPrimaryContactRoleMerge_CTRL() {
+        isRunningBatch = false;
+    }
+
+    /*********************************************************************************************************
+    * @description Returns the string Id of the panel. 
+    */
+    public override string idPanel() { return 'idPanelPrimaryContactRoleMergeBatch'; }
+
+    /*********************************************************************************************************
+    * @description Whether we are running the Remove Duplicate Primary OCRs Batch process.
+    */
+    public boolean isRunningBatch { get; set; }
+
+    /*********************************************************************************************************
+    * @description Action Method to run the Remove Duplicate Primary OCRs Batch.
+    * @return null 
+    */
+    public PageReference runBatch() {
+        isRunningBatch = true;
+        Database.executeBatch(new OPP_PrimaryContactRoleMerge_BATCH(), 2000);
```

Comment:
@kofijohnson Does it make sense to put the batch size into a Custom Settings field with a default of 2000? There will be a new UI page as part of the Customizable Rollups project to allow admins to define batch sizes for a bunch of different batch jobs. We can add this new field to that list so that admins will eventually be able to override the default size of 2000. @jjbennett

Reply:
Good point @force2b. However, in this case, the admin will not need to define the batch size. We want the size to be high as possible (2000) so the batch can go through the Opportunities as quick as possible.